packages/amplify-headless-interface/src/interface/auth/add.ts

```diff
@@ -187,6 +184,27 @@ export interface CognitoSocialProviderConfiguration {
   clientSecret: string;
 }
 
+interface SignInWithAppleSocialProviderConfig {
+  provider: 'SIGN_IN_WITH_APPLE';
+  /**
+   * The client ID (sometimes called apple services ID) configured with the provider.
+   */
+  clientId: string;
+
+  teamId: string;
```

Comment:
Add a comment to this prop and privateKey as these comments become the field description in the JSONSchema